frontend/src/v5/ui/routes/dashboard/projects/federations/editFederationModal/editFederation/editFederationContainersList/editFederationContainersListItem/groupOption/groupOption.component.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ *  Copyright (C) 2023 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { useContext } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { AddCircleIcon, Container, Message } from './groupOption.styles';
+import { EditFederationContext } from '../../../../editFederationContext';
+
+interface GroupOptionProps {
+	value: string;
+	onClick: (e) => void;
+}
+
+export const GroupOption = ({ value, ...props }: GroupOptionProps) => {
+	const { groups } = useContext(EditFederationContext);
+	const trimmedValue = value.trim();
+
+	const isNew = !groups.includes(trimmedValue);
+
+	if (!groups.length && !trimmedValue) {
+		return (
+			<Container {...props} disabled>
+				<Message>
+					<FormattedMessage id="tickets.groups.field.noOptions" defaultMessage="No options" />
+				</Message>
+			</Container>
+		);
+	}
+
+	if (!trimmedValue) return (<></>);
+
+	return (
```

Comment:
can we add a bit of a comment here stating that the css styles and behaviours from materialuitem will be applied through props?